### PR TITLE
ENH: refactoring code to fix internal compiler error in GCC 4.8

### PR DESCRIPTION
**Pull request by dzenanz**
Let's see whether this fixes CI's package problem.

### CODE REVIEW

File: include/itkMaxPhaseCorrelationOptimizer.hxx [hunk 1]
```diff
@@ -151,7 +151,7 @@ MaxPhaseCorrelationOptimizer< TRegistrationMethod >
 
   // supress trivial zero solution
   const typename ImageType::IndexType oIndex = lpr.GetIndex();
-  const PixelType                     zeroDeemphasis1 = std::max< PixelType >( 1.0, m_ZeroSuppression / 2.0 );
+  const PixelType zeroDeemphasis1 = std::max< PixelType >( 1.0, m_ZeroSuppression / 2.0 );
   for ( i = 0; i < maxs.size(); i++ )
```
Comment (hjmjohnson) [inline include/itkMaxPhaseCorrelationOptimizer.hxx]: i should have local scope to the for loop.  Not defined on line 116.  The variable on line 116 should be different than the variable at line 155.

Comment (dzenanz) [reply include/itkMaxPhaseCorrelationOptimizer.hxx]: OK I will split this into a style commit and an ENH

**Comment by thewtex** [comment]
Green CI! Way to go @dzenanz ! :taco: :tada:

**Comment by dzenanz** [comment]
There are many other warnings raised by GCC 4.8. It would be good if I have time to address them sometime in the future.

**Comment by dzenanz** [comment]
GCC warnings addressed in #84.

**Pull request merged**